### PR TITLE
OBSDOCS-1042&OBSDOCS-1048&OBSDOCS-1076: Make OTEL components show in TOC

### PR DESCRIPTION
**Pull request by max-cx**
:warning: Note for reviewers: you don't need to review all of the file lines because this PR introduces only the following changes:

- Split content currently on page https://docs.openshift.com/container-platform/4.14/observability/otel/otel-configuration-of-otel-collector.html into multiple separate pages (hence you see new files, but they contain already published content). This reorganization makes each component appear as a separate TOC entry in the right nav menu (currently only found by scrolling or Ctrl+F).
- Add `# ...` where relevant to YAML blocks (as requested in a previous peer review).
- Move the Technology Preview admonition boxes to appear after text paragraphs so that text appears first in each section (as requested in a previous peer review).

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

Issue:

https://issues.redhat.com/browse/OBSDOCS-1042
https://issues.redhat.com/browse/OBSDOCS-1048
https://issues.redhat.com/browse/OBSDOCS-1076

Link to docs preview:

https://77147--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-collector/otel-collector-connectors.html
https://77147--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-collector/otel-collector-exporters.html
https://77147--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-collector/otel-collector-extensions.html
https://77147--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-collector/otel-collector-processors.html
https://77147--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-collector/otel-collector-receivers.html
https://77147--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-collector/otel-collector-target-allocator.html

QE review:
- N/A (reorganized already published content)

Additional information:

### CODE REVIEW

File: observability/otel/otel-collector-exporters.adoc [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            server_name_override: kafka.example.corp # <7>
+    service:
+      pipelines:
+        traces:
+          exporters: [kafka]
+----
+<1> The list of Kafka brokers. The default is `+localhost:9092+`.
+<2> The Kafka protocol version. For example, `+2.0.0+`. This is a required field.
+<3> The name of the Kafka topic to read from. The following are the defaults: `+otlp_spans+` for traces, `+otlp_metrics+` for metrics, `+otlp_logs+` for logs.
+<4> The plaintext authentication configuration. If omitted, plaintext authentication is disabled.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply observability/otel/otel-collector-exporters.adoc]: Fixed.

[the remark this answers is not in the document]
Comment (max-cx) [reply observability/otel/otel-collector-exporters.adoc]: Fixed.

File: observability/otel/otel-collector-receivers.adoc [hunk 2]
```diff
@@ -0,0 +1,797 @@
...
+            server_name_override: kafka.example.corp # <7>
+    service:
+      pipelines:
+        traces:
+          receivers: [kafka]
+----
+<1> The list of Kafka brokers. The default is `+localhost:9092+`.
+<2> The Kafka protocol version. For example, `+2.0.0+`. This is a required field.
+<3> The name of the Kafka topic to read from. The default is `+otlp_spans+`.
+<4> The plaintext authentication configuration. If omitted, plaintext authentication is disabled.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply observability/otel/otel-collector-receivers.adoc]: Fixed.

[the remark this answers is not in the document]
Comment (max-cx) [reply observability/otel/otel-collector-receivers.adoc]: Fixed.

**Review by adellape — APPROVED** [review]
LGTM for the described scope of changes.

**Pull request merged**